src/os/bluestore/NVMEDevice.cc

```diff
@@ -587,41 +601,18 @@ static bool probe_cb(void *cb_ctx, struct spdk_pci_device *pci_dev)
     return false;
   }
 
-  if (spdk_pci_device_has_non_uio_driver(pci_dev)) {
-    /*NVMe kernel driver case*/
-    if (g_ceph_context->_conf->bdev_nvme_unbind_from_kernel) {
-      r =  spdk_pci_device_switch_to_uio_driver(pci_dev);
-      if (r < 0) {
-        derr << __func__ << " device " << name
-             << " " << spdk_pci_device_get_bus(pci_dev)
-             << ":" << spdk_pci_device_get_dev(pci_dev)
-             << ":" << spdk_pci_device_get_func(pci_dev)
-             << " switch to uio driver failed" << dendl;
-        return false;
-      }
-    } else {
-      derr << __func__ << " device has kernel nvme driver attached" << dendl;
-      return false;
-    }
-  } else {
-    r =  spdk_pci_device_bind_uio_driver(pci_dev);
-    if (r < 0) {
-      derr << __func__ << " device " << name
-           << " " << spdk_pci_device_get_bus(pci_dev)
-           << ":" << spdk_pci_device_get_dev(pci_dev) << ":"
-           << spdk_pci_device_get_func(pci_dev)
-           << " bind to uio driver failed, may lack of uio_pci_generic kernel module" << dendl;
-      return false;
-    }
-  }
-
   return true;
```

Comment:
The logic is in SPDK.  We will either use VFIO or UIO, the logic is shown in  scripts/setup.sh.  For pci_access,  the support is removed in current SPDK.